src/site/filters/liquid.js

```diff
@@ -68,4 +68,68 @@ module.exports = function registerFilters() {
     const id = `vha_${facilityLocatorApiId}`;
     return JSON.stringify(id);
   };
+
+  // Find the current path in an array of nested link arrays and then return it's depth + it's parent and children
+  liquid.filters.findCurrentPathDepth = (linksArray, currentPath) => {
+    const getDeepLinks = (path, linkArr) => {
+      const deepObj = {};
+      for (let a = 0; a < linkArr.length; a += 1) {
+        if (linkArr[a].url.path === path) {
+          deepObj.depth = 1;
+          return deepObj;
+        }
+        if (linkArr[a].links) {
+          for (let b = 0; b < linkArr[a].links.length; b += 1) {
+            if (linkArr[a].links[b].url.path === path) {
+              deepObj.depth = 2;
+              return deepObj;
+            }
+            if (linkArr[a].links[b].links) {
+              for (let c = 0; c < linkArr[a].links[b].links.length; c += 1) {
+                if (linkArr[a].links[b].links[c].url.path === path) {
+                  deepObj.depth = 3;
+                  deepObj.links = linkArr[a].links[b];
+                  return deepObj;
+                }
+                if (linkArr[a].links[b].links[c].links) {
+                  for (
+                    let d = 0;
+                    d < linkArr[a].links[b].links[c].links.length;
+                    d += 1
+                  ) {
+                    if (
+                      linkArr[a].links[b].links[c].links[d].url.path === path
+                    ) {
+                      deepObj.depth = 4;
+                      deepObj.links = linkArr[a].links[b].links[c];
+                      return deepObj;
+                    }
+                    if (linkArr[a].links[b].links[c].links[d].links) {
+                      for (
+                        let e = 0;
+                        e < linkArr[a].links[b].links[c].links[d].links.length;
+                        e++
+                      ) {
+                        if (
+                          linkArr[a].links[b].links[c].links[d].links[e].url
+                            .path === path
+                        ) {
+                          deepObj.depth = 5;
+                          deepObj.links = linkArr[a].links[b].links[c].links[d];
+                          return deepObj;
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+      return false;
+    };
+
+    return JSON.stringify(getDeepLinks(currentPath, linksArray));
```

Comment:
Does liquid force filters to return strings? Or could you return the object here?